File: MDS_project/MDS_frontend/src/app/interfaces/musical-band.ts
import { UserProfile } from "./user-profile"
import { BandHQ } from "./band-hq"
import { BandAndUserMatch } from "./band-user-match"

export interface MusicalBand {
    bandId: number
    name: string
    musicGenre?: string
    dateFormed: Date
    isComplete: boolean
    members?: UserProfile[]
    hq?: BandHQ

    bandAndUserMatches?: BandAndUserMatch[]
}

File: MDS_project/MDS_backend/MDS_backend/Entities/MusicalBand.cs
using System.ComponentModel.DataAnnotations;

namespace MDS_backend.Entities
{
    public class MusicalBand
    {
        [Required, Key]
        public int BandId { get; set; }

        [Required]
        public string Name { get; set; } = null!;
        public string? MusicGenre { get; set; }
        public DateTime DateFormed { get; set; }
        public bool IsComplete { get; set; }

        public virtual BandHQ? HQ { get; set; }
        public virtual ICollection<UserProfile> Members { get; set; } = null!;

    }
}
